packages/saltcorn-data/models/table.ts

```diff
@@ -3706,13 +3718,19 @@ ${rejectDetails}`,
    */
   async getJoinedQuery(
     opts: (JoinOptions & ForUserRequest) | any = {}
-  ): Promise<any> {
+  ): Promise<
+    Partial<JoinOptions> & {
+      sql?: string;
+      values?: any[];
```

Comment:
We have a type `Value` in db-common/internal.ts you could try to use here if not too hard